### ISSUE TITLE
ucp_tag_send_nbr() does not use calculated rendezvous threshold

### ISSUE THREAD

**Issue by souravzzz** [openucx/ucx#4430]
In both `ucp_ep_config_set_rndv_thresh()` and `ucp_ep_config_set_am_rndv_thresh()`, we have

```
rndv_thresh     = ucp_ep_config_calc_rndv_thresh(...);
rndv_nbr_thresh = context->config.ext.rndv_send_nbr_thresh;
```

Consequently, `ucp_tag_send_nb()` uses a rendezvous threshold calculated based on the interface characteristics while `ucp_tag_send_nbr()` uses a hard-coded value of 256K. Since MPI libraries use `ucp_tag_send_nbr()` if available, rendezvous thresholds are not getting automatically calculated. Is this behavior expected?

**Comment by brminich** [comment]
Yes, it is expected. MPI libraries use `ucp_tag_send_nb()` also. 
`ucp_tag_send_nbr()` mostly used for blocking MPI_Send()

**Comment by souravzzz** [comment]
@brminich Indeed. But doesn't this mean that blocking MPI sends will always use a rendezvous threshold of 256K regardless of the performance characteristics of the underlying transport, unless overridden by the environment variable? Is there any particular reason this value was chosen?
